### ISSUE TITLE
Upsert operation doesn't return the expected status and/or body

### ISSUE THREAD

**Issue by kiskit** [Fiware/context.Orion-LD#1221]
This issue seems to relate to issue #805 .
## Problem statement
A successful upsert operation on both existing and non existing entities does not return the correct body and/or status.
## Modus Operandi
First create an entity such as this one:
```json
[
	{
		"id": "uri:entity1",
		"type": "Temperature",
		"dataProvider": {
			"type": "Property",
			"value": "62349895462c09197b1d6776"
		},
		"xy": {
			"type": "Property",
			"value": "apopo"
		}
	}
]
```
Then use the upsert operation (/entityOperations/upsert) to modify the entity previously created and insert another one:
```json
[
	{
		"dataProvider": {
			"type": "Property",
			"value": "62349895462c09197b1d6776"
		},
		"id": "uri:entity1",
		"type": "Temperature"
	},
	{
		"dataProvider": {
			"type": "Property",
			"value": "62349895462c09197b1d6776"
		},
		"id": "uri:entity2",
		"type": "Temperature"
	}
]
```
## Expectations
In version "latest" or 1.1.0 (as pulled from the docker repository), the status returned is **207** and the body is a **BatchOperationResult**:
```json 
{
  "success": [
    "uri:entity1",
    "uri:entity2"
  ],
  "errors": []
}
```
However, according to the specification (chapter 6.15), that case should warrant a **201** status, with a **list of IDs of all the entities created and updated** returned in the body.

**Comment by kzangeli** [comment]
Yes, you're right, the broker isn't up-to-date with the latest changes in the spec.

However, I'm rewriting part of the service routines to use the new mongo driver (mongoc) instead of the older legacy C++ driver, that was stamped as obsolete quite some time ago.

All batch ops are reimplemented and they follow the spec (returning a 201 and an array of entity ids - those entities that were CREATED only).

If you want to check it out, start the broker with `-experimental`.

Fixing old obsolete stuff is not very high up on my ToDo ...

Anyway, thanks for reporting!!!

**Comment by kiskit** [comment]
It's no biggie on our side for now to be honest. Good point about the created thing, I had misread the spec. Thanks
